load-test-framework/src/main/java/com/google/pubsub/flic/Driver.java

```diff
@@ -15,124 +15,236 @@
-          taskArgs = builder.broker(kafkaArgs.getBroker()).build();
-          kafkaHandler.setLatencyType(MessageProcessingHandler.LatencyType.END_TO_END);
-          KafkaConsumer<String, String> consumer =
-              KafkaConsumerTask.getInitializedConsumer(taskArgs);
-          log.info("Creating a task which consumes from Kafka.");
-          new KafkaConsumerTask(taskArgs, consumer, kafkaHandler).execute();
+      Client.messageSize = messageSize;
+      Client.requestRate = 1;
+      Client.startTime = Timestamp.newBuilder().setSeconds(System.currentTimeMillis() / 1000 + 90).build();
+      Client.loadtestLengthSeconds = loadtestLengthSeconds;
+      Client.batchSize = batchSize;
+      Client.maxMessagesPerPull = cpsMaxMessagesPerPull;
+      Client.pollLength = kafkaPollLength;
+      Client.broker = broker;
+      Client.requestRate = requestRate;
+      Client.maxOutstandingRequests = maxOutstandingRequests;
+      Client.burnInTimeMillis = (Client.startTime.getSeconds() + burnInDurationSeconds) * 1000;
+      Client.numberOfMessages = numberOfMessages;
+      GCEController gceController = GCEController.newGCEController(
+          project, ImmutableMap.of("us-central1-a", clientParamsMap), Executors.newScheduledThreadPool(500));
+      gceController.startClients();
+
+      // Start a thread to poll and output results.
+      ScheduledExecutorService pollingExecutor = Executors.newSingleThreadScheduledExecutor();
+      pollingExecutor.scheduleWithFixedDelay(() -> {
+        synchronized (pollingExecutor) {
+          log.info("===============================================");
+          printStats(gceController.getResults());
+          log.info("===============================================");
         }
+      }, 5, 5, TimeUnit.SECONDS);
+      // Wait for the load test to finish.
+      gceController.waitForClients();
+      synchronized (pollingExecutor) {
+        pollingExecutor.shutdownNow();
       }
-    } catch (Exception e) {
-      log.error("An error occurred...", e);
+      printStats(gceController.getResults());
+
+      gceController.shutdown(null);
+      System.exit(0);
+    } catch (Throwable t) {
+      log.error("An error occurred...", t);
       System.exit(1);
     }
   }
+
+  private void printStats(Map<ClientType, Controller.Result> results) {
+    results.forEach((type, result) -> {
+      log.info("Results for " + type + ":");
+      log.info("50%: " + LatencyDistribution.getNthPercentile(result.bucketValues, 0.5));
+      log.info("99%: " + LatencyDistribution.getNthPercentile(result.bucketValues, 0.99));
+      log.info("99.9%: " + LatencyDistribution.getNthPercentile(result.bucketValues, 0.999));
+      // CPS Publishers report latency per batch message.
+      log.info("Average throughput: " +
+          new DecimalFormat("#.##").format(
+              (double) LongStream.of(
+                  result.bucketValues).sum() / (result.endTimeMillis / 1000.0 - Client.startTime.getSeconds())
```

Comment:
Do we still need to rely on Client.startTime if we have running_time from the CheckResponse?